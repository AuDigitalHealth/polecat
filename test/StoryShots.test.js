import initStoryshots from '@storybook/addon-storyshots'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { Route } from 'react-router'
import React from 'react'

const fakeStore = {
  getState: () => {},
  dispatch: () => null,
  subscribe: () => null,
  replaceReducer: () => null,
}

// FIXME: Error when running Storybook:
// Could not find "store" in either the context or props of
// "Connect(RemoteFhirMedication)". Either wrap the root component in a
// <Provider>, or explicitly pass "store" as a prop to
// "Connect(RemoteFhirMedication)".
const shallowSnapshot = ({ story, context }) => {
  const element = story.render(context)
  const options = { disableLifecycleMethods: true }
  let result = element
  // Skip any wrapper elements added by decorators.
  while (result.props.story) result = result.props.story()
  // If component wants to be connected to a Redux store, give it a fake one.
  if (result.type.displayName && result.type.displayName.match(/^Connect/)) {
    result = React.cloneElement(result, { store: fakeStore })
    result = shallow(result, options)
  } else {
    result = shallow(result, options)
  }
  // Skip any `Route` element added by story-router.
  const route = result.find(Route)
  if (route.exists()) result = shallow(route.prop('render')(), options)
  expect(result).toMatchSnapshot()
}

Enzyme.configure({ adapter: new Adapter() })
initStoryshots({ test: options => shallowSnapshot(options) })
