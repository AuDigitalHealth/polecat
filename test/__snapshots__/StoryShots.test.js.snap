// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots AmtBrowser Codalgin Forte uncoated tablet, 20, blister pack (inactive) 1`] = `
<div
  className="amt-browser"
  style={
    Object {
      "height": 800,
      "width": 800,
    }
  }
>
  <Connect(RemoteFhirMedication)
    id="19255011000036102"
    onError={[Function]}
    onLoadSubjectConcept={[Function]}
    onLoadingChange={[Function]}
    resourceType="Medication"
    viewport={
      Object {
        "height": 800,
        "width": 800,
      }
    }
  >
    <Connect(AmtProductModel)
      viewport={
        Object {
          "height": 800,
          "width": 800,
        }
      }
    />
  </Connect(RemoteFhirMedication)>
  <Connect(withRouter(Search))
    focusUponMount={true}
    onError={[Function]}
  />
  <Loading
    delay={350}
    loading={false}
  />
</div>
`;

exports[`Storyshots AmtBrowser Codalgin Forte uncoated tablet, 20, blister pack (replaces inactive) 1`] = `
<div
  className="amt-browser"
  style={
    Object {
      "height": 800,
      "width": 800,
    }
  }
>
  <Connect(RemoteFhirMedication)
    id="835891000168108"
    onError={[Function]}
    onLoadSubjectConcept={[Function]}
    onLoadingChange={[Function]}
    resourceType="Medication"
    viewport={
      Object {
        "height": 800,
        "width": 800,
      }
    }
  >
    <Connect(AmtProductModel)
      viewport={
        Object {
          "height": 800,
          "width": 800,
        }
      }
    />
  </Connect(RemoteFhirMedication)>
  <Connect(withRouter(Search))
    focusUponMount={true}
    onError={[Function]}
  />
  <Loading
    delay={350}
    loading={false}
  />
</div>
`;

exports[`Storyshots AmtBrowser Lorapaed 1 mg/mL oral liquid solution (TPUU) 1`] = `
<div
  className="amt-browser"
  style={
    Object {
      "height": 800,
      "width": 800,
    }
  }
>
  <Connect(RemoteFhirMedication)
    id="53743011000036103"
    onError={[Function]}
    onLoadSubjectConcept={[Function]}
    onLoadingChange={[Function]}
    resourceType="Medication"
    viewport={
      Object {
        "height": 800,
        "width": 800,
      }
    }
  >
    <Connect(AmtProductModel)
      viewport={
        Object {
          "height": 800,
          "width": 800,
        }
      }
    />
  </Connect(RemoteFhirMedication)>
  <Connect(withRouter(Search))
    focusUponMount={true}
    onError={[Function]}
  />
  <Loading
    delay={350}
    loading={false}
  />
</div>
`;

exports[`Storyshots AmtBrowser Lorapaed 1 mg/mL oral liquid solution, 150 mL (TPP) 1`] = `
<div
  className="amt-browser"
  style={
    Object {
      "height": 800,
      "width": 800,
    }
  }
>
  <Connect(RemoteFhirMedication)
    id="55971011000036104"
    onError={[Function]}
    onLoadSubjectConcept={[Function]}
    onLoadingChange={[Function]}
    resourceType="Medication"
    viewport={
      Object {
        "height": 800,
        "width": 800,
      }
    }
  >
    <Connect(AmtProductModel)
      viewport={
        Object {
          "height": 800,
          "width": 800,
        }
      }
    />
  </Connect(RemoteFhirMedication)>
  <Connect(withRouter(Search))
    focusUponMount={true}
    onError={[Function]}
  />
  <Loading
    delay={350}
    loading={false}
  />
</div>
`;

exports[`Storyshots AmtBrowser Lorapaed 1 mg/mL oral liquid solution, 150 mL, bottle (CTPP) 1`] = `
<div
  className="amt-browser"
  style={
    Object {
      "height": 800,
      "width": 800,
    }
  }
>
  <Connect(RemoteFhirMedication)
    id="60009011000036108"
    onError={[Function]}
    onLoadSubjectConcept={[Function]}
    onLoadingChange={[Function]}
    resourceType="Medication"
    viewport={
      Object {
        "height": 800,
        "width": 800,
      }
    }
  >
    <Connect(AmtProductModel)
      viewport={
        Object {
          "height": 800,
          "width": 800,
        }
      }
    />
  </Connect(RemoteFhirMedication)>
  <Connect(withRouter(Search))
    focusUponMount={true}
    onError={[Function]}
  />
  <Loading
    delay={350}
    loading={false}
  />
</div>
`;

exports[`Storyshots AmtBrowser Search for esomeprazole 1`] = `
<div
  className="amt-browser"
  style={
    Object {
      "height": 800,
      "width": 800,
    }
  }
>
  <Connect(withRouter(Search))
    focusUponMount={true}
    onError={[Function]}
    query="esomeprazole"
  />
  <Loading
    delay={350}
    loading={false}
  />
</div>
`;

exports[`Storyshots AmtBrowser loratadine (MP) 1`] = `
<div
  className="amt-browser"
  style={
    Object {
      "height": 800,
      "width": 800,
    }
  }
>
  <Connect(RemoteFhirMedication)
    id="21701011000036106"
    onError={[Function]}
    onLoadSubjectConcept={[Function]}
    onLoadingChange={[Function]}
    resourceType="Medication"
    viewport={
      Object {
        "height": 800,
        "width": 800,
      }
    }
  >
    <Connect(AmtProductModel)
      viewport={
        Object {
          "height": 800,
          "width": 800,
        }
      }
    />
  </Connect(RemoteFhirMedication)>
  <Connect(withRouter(Search))
    focusUponMount={true}
    onError={[Function]}
  />
  <Loading
    delay={350}
    loading={false}
  />
</div>
`;

exports[`Storyshots AmtBrowser loratadine (substance) 1`] = `
<div
  className="amt-browser"
  style={
    Object {
      "height": 800,
      "width": 800,
    }
  }
>
  <Connect(RemoteFhirMedication)
    id="2292011000036106"
    onError={[Function]}
    onLoadSubjectConcept={[Function]}
    onLoadingChange={[Function]}
    resourceType="Substance"
    viewport={
      Object {
        "height": 800,
        "width": 800,
      }
    }
  >
    <Connect(AmtProductModel)
      viewport={
        Object {
          "height": 800,
          "width": 800,
        }
      }
    />
  </Connect(RemoteFhirMedication)>
  <Connect(withRouter(Search))
    focusUponMount={true}
    onError={[Function]}
  />
  <Loading
    delay={350}
    loading={false}
  />
</div>
`;

exports[`Storyshots AmtBrowser loratadine 1 mg/mL oral liquid (MPUU) 1`] = `
<div
  className="amt-browser"
  style={
    Object {
      "height": 800,
      "width": 800,
    }
  }
>
  <Connect(RemoteFhirMedication)
    id="62022011000036104"
    onError={[Function]}
    onLoadSubjectConcept={[Function]}
    onLoadingChange={[Function]}
    resourceType="Medication"
    viewport={
      Object {
        "height": 800,
        "width": 800,
      }
    }
  >
    <Connect(AmtProductModel)
      viewport={
        Object {
          "height": 800,
          "width": 800,
        }
      }
    />
  </Connect(RemoteFhirMedication)>
  <Connect(withRouter(Search))
    focusUponMount={true}
    onError={[Function]}
  />
  <Loading
    delay={350}
    loading={false}
  />
</div>
`;

exports[`Storyshots AmtBrowser loratadine 1 mg/mL oral liquid, 150 mL (MPP) 1`] = `
<div
  className="amt-browser"
  style={
    Object {
      "height": 800,
      "width": 800,
    }
  }
>
  <Connect(RemoteFhirMedication)
    id="63589011000036105"
    onError={[Function]}
    onLoadSubjectConcept={[Function]}
    onLoadingChange={[Function]}
    resourceType="Medication"
    viewport={
      Object {
        "height": 800,
        "width": 800,
      }
    }
  >
    <Connect(AmtProductModel)
      viewport={
        Object {
          "height": 800,
          "width": 800,
        }
      }
    />
  </Connect(RemoteFhirMedication)>
  <Connect(withRouter(Search))
    focusUponMount={true}
    onError={[Function]}
  />
  <Loading
    delay={350}
    loading={false}
  />
</div>
`;
